handler/src/main/java/io/netty/handler/ssl/SslUtils.java

```diff
@@ -293,13 +298,17 @@ static int getEncryptedPacketLength(ByteBuf buffer, int offset) {
         if (tls) {
             // SSLv3 or TLS or GMSSLv1.0 or GMSSLv1.1 - Check ProtocolVersion
             int majorVersion = buffer.getUnsignedByte(offset + 1);
+            int version = buffer.getShort(offset + 1);
             if (majorVersion == 3 || buffer.getShort(offset + 1) == GMSSL_PROTOCOL_VERSION) {
                 // SSLv3 or TLS or GMSSLv1.0 or GMSSLv1.1
                 packetLength = unsignedShortBE(buffer, offset + 3) + SSL_RECORD_HEADER_LENGTH;
                 if (packetLength <= SSL_RECORD_HEADER_LENGTH) {
                     // Neither SSLv3 or TLSv1 (i.e. SSLv2 or bad data)
                     tls = false;
                 }
+            } else if (version == DTLS_1_0 || version == DTLS_1_2 || version == DTLS_1_3) {
+                // length is the last 2 bytes in the 13 byte header. 13 - 2 = 11
+                packetLength = unsignedShortBE(buffer, offset + 11) + DTLS_RECORD_HEADER_LENGTH;
```

Comment:
I dont think this is save as there might be not enough readable bytes in the buffer.
```suggestion
                int packetLengthOffset = offset + 11;
                if (buffer.readableBytes() < packetLengthOffset + 2) {
                    return NOT_ENOUGH_DATA;
                }
                packetLength = unsignedShortBE(buffer, packetLengthOffset) + DTLS_RECORD_HEADER_LENGTH;
```

Reply:
🤦 Yep. I went ham deleting. Will fix. Still need the check just not the while loop.

Reply:
@normanmaurer Fixed this and also added a test to verify if the length isn't long enough we return NOT_ENOUGH_DATA and fixed the test with two DTLS records.

Reply:
Actually I think we dont need this change at all as we check `DTLS_RECORD_HEADER_LENGTH` before.